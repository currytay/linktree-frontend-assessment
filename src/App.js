import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import { Header } from './components/header';

const App = () => {

  return (
    <AppContainer>
      <ProfileContainer>
        <Header />
      </ProfileContainer>
    </AppContainer>
  );
};

const AppContainer = styled.div`
  width: 100%;
`;

const ProfileContainer = styled.div`
  width: 100%;
  max-width: 42.5rem;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem 1.5rem;
`;

export default App;