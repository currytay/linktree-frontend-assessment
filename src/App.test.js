import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing', () => {
  shallow(<App />);
});

it('renders user handle', () => {
  const profile = shallow(<App />);
  const handle = <h1>@yourname</h1>;
  expect(profile.contains(handle));
});

test('snapshot renders', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});