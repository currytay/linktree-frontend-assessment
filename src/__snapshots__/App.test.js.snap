// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="sc-bBHxTw erjYgj"
>
  <div
    className="sc-iwjdpV fxldqT"
  >
    <div
      className="sc-dkPtRN ciIKFR"
    >
      <img
        alt="User profile avatar"
        className="sc-bdvvtL kaSKQS"
        src="profile-picture.png"
      />
      <h1
        className="sc-gsDKAQ dABKQ"
      >
        @yourname
      </h1>
    </div>
    <div
      className="sc-cxpSdN dqjbpH"
    />
    <div
      className="sc-llYSUQ iQJvvR"
    >
      <img
        alt="Linktree logo"
        src="logo.svg"
      />
    </div>
  </div>
</div>
`;
